### PR TITLE
Implement manual domain set in Proxy Builder

### PR DESCRIPTION
**Pull request by mhabedini**

### CODE REVIEW

File: src/Requests/PendingRequest.php [hunk 1]
```diff
@@ -74,7 +83,7 @@ public function patch($url, $data = [])
         return $this->respond($url, $data, Request::METHOD_PATCH);
     }
 
-    public function post(string $url, array $data = [])
```
Comment (omalizadeh) [inline src/Requests/PendingRequest.php:77]: why this array type removed?

File: tests/Feature/ProxyTest.php [hunk 2]
```diff
@@ -45,6 +45,20 @@ public function testFailedProxyRequest(): void
...
+    {
+        $domain = 'http://sub.domain.example';
+        Proxy::fake([
+            $domain.'/test-service/api/path/1' => Http::response(['foo' => 'bar']),
+        ]);
+
+        Proxy::domain($domain)->get('test-service/api/path/1')
+            ->onSuccess(function (ProxyResponse $responseWrapper) {
+                assertEquals(['foo' => 'bar'], $responseWrapper->response()->json());
+                assertEquals('/test-service/api/path/1', $responseWrapper->response()->effectiveUri()->getPath());
```
Comment (omalizadeh) [inline tests/Feature/ProxyTest.php:58]: i think we assert path in other tests. you should asset that domain changed in this test.

**Review by omalizadeh — CHANGES_REQUESTED** [review]
checkout comments

File: src/Requests/PendingRequest.php (hunk 1 above)
Comment (mhabedini) [reply src/Requests/PendingRequest.php:77]: Because the post method data argument in superclass has no type.

**Review by omalizadeh — APPROVED** [review]

**Pull request merged**